screens/productScreen/ChooseColorModal/ChooseColorModal.js

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from "react";
+import { Modal, Pressable, Text, View } from "react-native";
+import { Ionicons } from "@expo/vector-icons";
+import { styles } from "../sizeModalScreen/SizeModalScreen";
+
+const ColorModalScreen = ({ colors }) => {
+  const [modalVisibility, setModalVisibility] = useState(false);
+  const [selectedColor, setSelectedColor] = useState("Color");
+  const [colorPressed, setColorPressed] = useState(false);
+
+  const handleColorSelection = (color) => {
+    setSelectedColor(color);
+    setModalVisibility(false); // Close the modal when a color is selected
+  };
+
+  return (
+    <View style={styles.centeredView}>
+      <Pressable
+        style={[styles.choosing, colorPressed && styles.choosingPressed]}
+        onPress={() => {
+          setColorPressed(!colorPressed);
+          setModalVisibility(true);
+        }}
+      >
+        {selectedColor !== "Color" ? (
+          <View
+            style={{ width: 20, height: 20, backgroundColor: selectedColor }}
```

Comment:
add style

Reply:
i couldn't find way to pass the selectedColor to style if im not using inline style. i tried to sepperate widh and height from background color and use inline style only for the background and it didn't work

Reply:
never mind, i fixed it :)